tests/phpunit/integration/Core/Authentication/AuthenticationTest.php

```diff
@@ -235,6 +242,134 @@ public function test_register_allowed_redirect_hosts() {
+	}
+
+	public function test_register_maybe_refresh_token_for_screen__admin_without_shared_modules() {
+		$this->enable_feature( 'dashboardSharing' );
+		$this->test_register_maybe_refresh_token_for_screen__admin();
+	}
+
+	public function test_register_maybe_refresh_token_for_screen__editor_with_shared_modules() {
+		$this->enable_feature( 'dashboardSharing' );
+		$editor_id = $this->factory()->user->create( array( 'role' => 'editor' ) );
+		wp_set_current_user( $editor_id );
+		$context      = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$user_options = new User_Options( $context, $editor_id );
+
+		// Allow PSI to be shared with an editor.
+		$test_sharing_settings = array(
+			'pagespeed-insights' => array(
+				'sharedRoles' => array( 'editor' ),
+				'management'  => 'all_admins',
+			),
+			'search-console'     => array(
+				'management' => 'owner',
+			),
+		);
+		add_option( Module_Sharing_Settings::OPTION, $test_sharing_settings );
+
+		$auth = new Authentication( $context, null, $user_options );
+		$auth->register();
+
+		// Re-register Permissions after enabling the dashboardSharing feature to include dashboard sharing capabilities.
+		$permissions = new Permissions( $context, $auth, new Modules( $context ), $user_options, new Dismissed_Items( $user_options ) );
+		$permissions->register();
+
+		$oauth_client = $auth->get_oauth_client();
+		// The FakeHttpClient returns 200 by default.
+		$oauth_client->get_client()->setHttpClient( new FakeHttpClient() );
+		$user_options->set( OAuth_Client::OPTION_ACCESS_TOKEN_EXPIRES_IN, 295 );
+
+		// Create owners for shareable modules and generate their oauth tokens.
+		$pagespeed_insights_owner_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+		update_option( PageSpeed_Insights_Settings::OPTION, array( 'ownerID' => $pagespeed_insights_owner_id ) );
+		$this->set_oauth_token_for_user( $oauth_client, $user_options, $pagespeed_insights_owner_id );
+		$search_console_owner_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+		update_option( Search_Console_Settings::OPTION, array( 'ownerID' => $search_console_owner_id ) );
+		$this->set_oauth_token_for_user( $oauth_client, $user_options, $search_console_owner_id );
+
+		// Make sure we start with no errors.
+		$this->assertFalse( get_user_option( OAuth_Client::OPTION_ERROR_CODE, $editor_id ) );
+		$this->assertFalse( get_user_option( OAuth_Client::OPTION_ERROR_CODE, $pagespeed_insights_owner_id ) );
+		$this->assertFalse( get_user_option( OAuth_Client::OPTION_ERROR_CODE, $search_console_owner_id ) );
+
+		do_action( 'current_screen', WP_Screen::get( 'toplevel_page_googlesitekit-dashboard' ) );
+
+		// Token should not be refreshed for the editor who is not an authenticated admin.
+		$this->assertFalse( get_user_option( OAuth_Client::OPTION_ERROR_CODE, $editor_id ) );
+		// There is no actual response when refresh_token() is executed - so this is a successful call to refresh_token().
```

Comment:
Again, let's explain why this is and reference it in the issue created 😄 

```suggestion
		// There is no actual response when refresh_token() is executed - so this is a successful call to refresh_token().
```